src/visualization/ColorPalette/ColorMap.cc

```diff
@@ -0,0 +1,284 @@
+    int array[60];
+    for (int i=0;i<60;i++)
+        array[i] = 11 + i;
+    for (int l=59;l>=0;l--) {
+        for (int i=0;i<4;i++) {
+            for (int j=0;j<4;j++) {
+                for (int k=0;k<4;k++) {
+                    if (index==numbOfKmer) {
+                        l = -1;
+                        i = 4;
+                        j = 4;
+                        break;
+                    }
+                    index += 1;
+                    placeholder.r = (((i)*60)+array[l]);
+                    placeholder.g = (((j)*60)+array[l]);
+                    placeholder.b = (((k)*60)+array[l]);
+                    returnList.push_back(placeholder);
+                }
+            }
+        }
+    }
+    ColorList = returnList;
+}
+
+ColorMap::ColorMap(){
+
+}
+
+/**
+ * This Method will compute in terms of the argument sequences and the k, all kmers we are looking for and call
+ * the method which compute us in terms of this number a Colormap. Then the method will compute
+ * all different Kmers and push them in a List. We'll use this list to update differentKmers.
+ */
+void ColorMap::updateSequenceLists(vector<string>& Sequences, int k){
+    ///< This is a variable which save the number of Kmers per Sequence which we have
+    vector<int> SequenceSizes;
+    for (int i=0;i<Sequences.size();i++)
+        SequenceSizes[i] = (int)floor((Sequences[i].size())/k);
+
+    ///< This is a variable which save the number of Kmers
+    int numbOfKmer = 0;
+    for (int i=0;i<SequenceSizes.size();i++)
+        numbOfKmer += SequenceSizes[i];
+    updateColorList(numbOfKmer);
+    ///< This variable is the output List
+    vector<string> kmerDeclarations;
+
+    ///< a boolean for the "got this kmer a color?"-Check
+    bool matched = false;
+    ///< a index for the k-mers per sequence
+    int index;
+    
+    /**
+     * This part of the method will push all Kmers in the List, which are not in the list already.
+     */
+    for (int i=0;i<Sequences.size();i++) {
+        index = 0;
+        while (index<SequenceSizes[i]) {
+            ///< Comparing all k-mers to know if the Kmer is already in the list
+            for (int j=0;j<kmerDeclarations.size();j++) {
+                if (kmerDeclarations[j]==Sequences[i].substr(index*k, k)) {
+                    matched = true;
+                    break;
+                }
+            }
+            ///< If its a new kmer, so we push it here
+            if (!matched)
+                kmerDeclarations.push_back(Sequences[i].substr(index*k, k));
+            matched = false;
+            index += 1;
+        }
+    }
+    differentKmers = kmerDeclarations;
+}
+
+/**
+ * This Method will make a vector of Colors in terms of a number, which defines the Length of the 
+ * length of the List. Then it will update our ColorList with it. In our case the argument 
+ * describes the number of Kmers we are looking for to map.
+ */
+void ColorMap::updateColorList(int numbOfKmer){
```

Comment:
This is too much duplicated code. There needs to be a single function. It may be called in different ways or using different wrappers.